solaris/admin/create_gpt_parts.sh

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env bash
+# =============================================================================
+# Solaris GPT Partition Creator - OpenIndiana/illumos System Layout
+# -----------------------------------------------------------------------------
+# Purpose : Create GPT partition layout optimized for OpenIndiana/illumos systems.
+#           Sets up ESP, Solaris root ZFS pool, data partition, and reserved partition.
+# Usage   : Run after disk_hammer_illumos.sh to create clean partition layout.
+#           Set DISK_DEVICE variable to your target disk.
+# Notes   : - Requires prior run of disk_hammer_illumos.sh for clean disk state
+#           - Creates 4 partitions: ESP (512MB), Solaris root, data, reserved (8MB)
+#           - Uses proper Solaris partition type codes (BF00, BF05, BF07)
+# =============================================================================
+
+# Configuration: Set your target disk device here
+# Examples: sda, sdb, nvme0n1, nvme1n1, etc.
+DISK_DEVICE="sdX"  # CHANGE THIS TO YOUR ACTUAL DISK
```

Comment:
This is the point where you let user to specify command line arguments (user should not update the script to work with disk). 

consider this example:
while getopts d:; do
  case $d in
  'd')
       # do you want to add path to device there, so you wont need to use "/dev" all over the script?
       [[ -c $OPTARG ]] || fail "no such device $OPTARG"
       DISK_DEVICE=$OPTARG
       ;;
       esac
done
shift $((OPTIND - 1))

(see getopts(1)) for more examples).